tests/integration/nm/linux_bridge_test.py

```diff
@@ -101,22 +123,47 @@ def _bridge_interface(state):
             _delete_iface(p[LB.PORT_NAME])
 
 
+@mainloop_run
+def _modify_bridge(bridge_desired_state):
+    bridge_config = bridge_desired_state[LB.CONFIG_SUBTREE]
+    _modify_bridge_options(bridge_config)
+    _modify_ports(bridge_config[LB.PORT_SUBTREE])
+
+
+def _modify_bridge_options(bridge_config):
+    br_options = bridge_config.get(LB.OPTIONS_SUBTREE)
+    nmdev = nm.device.get_device_by_name(BRIDGE0)
+    conn = nm.connection.ConnectionProfile()
+    conn.import_by_id(BRIDGE0)
+    iface_bridge_settings = _create_iface_bridge_settings(br_options, conn)
+    new_conn = nm.connection.ConnectionProfile()
+    new_conn.create(iface_bridge_settings)
+    conn.update(new_conn)
+    conn.commit(save_to_disk=False, nmdev=nmdev)
+    nm.device.modify(nmdev, conn.profile)
+
+
+def _modify_ports(ports_state):
+    for port_state in ports_state:
+        _attach_port_to_bridge(port_state)
+
+
 def _get_bridge_current_state():
     nm.nmclient.client(refresh=True)
     nmdev = nm.device.get_device_by_name(BRIDGE0)
     return nm.bridge.get_info(nmdev) if nmdev else {}
 
 
+@mainloop_run
 def _create_bridge(bridge_desired_state):
     bridge_config = bridge_desired_state.get(LB.CONFIG_SUBTREE, {})
     br_options = bridge_config.get(LB.OPTIONS_SUBTREE)
-    iface_bridge_settings = _get_iface_bridge_settings(br_options)
+    iface_bridge_settings = _create_iface_bridge_settings(br_options)
 
-    with mainloop():
-        _create_bridge_iface(iface_bridge_settings)
-        ports_state = bridge_desired_state[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE]
-        for port_state in ports_state:
-            _attach_port_to_bridge(port_state)
+    _create_bridge_iface(iface_bridge_settings)
+    ports_state = bridge_desired_state[LB.CONFIG_SUBTREE][LB.PORT_SUBTREE]
+    for port_state in ports_state:
+        _attach_port_to_bridge(port_state)
 
 
 def _attach_port_to_bridge(port_state):
```

Comment:
feels very weird that this does not receive the bridge as well, but if I recall your words correctly, adding a port to a bridge, is setting the port to be a slave of bridge - hence I think it would be a property of the bridge. 

Am I right?

Reply:
Yes, you are right. However, this refactoring was aimed to serve the following patch and not fix all issues.
The problem is with the usage of `BRIDGE0` all over the place. It should have been used only at the top most level (test) and not in the helpers.
I would prefer not to add this task on this change, but I think it is reasonable to add a patch that fixes it in general.